NUbook Page on Mocap (#339)
NUbots/NUbook: src/book/03-guides/02-tools/06-mocap.mdx

---
section: Guides
chapter: Tools
title: Setting Up Motion Capture
description: Setup instructions for Motive 2.3.1
slug: /guides/tools/mocap-setup
authors:
  - Will Burgin (@willburgin)
---

[Motive](https://optitrack.com/software/motive/) is the software package used to run our OptiTrack motion capture system. This guide explains how to set up and operate the system using **Motive 2.3.1**. By the end of this guide, you should know how to use the Motion Capture System to obtain ground truth data.

---

## Prerequisites

<details>

<summary>Hardware</summary>

> - **Dedicated MSI Laptop** (Windows 10, Motive 2.3.1 pre-installed)

Reviewer: Maybe make this more generic but say that as of writing, the MSI laptop is set up with Motive in the lab.
eg
'Computer with Windows 10, Motive 2.3.1, [any port requirements, eg ethernet] - MSI laptop in the NUbots lab has these requirements'

- **Motive License USB dongle** (must be plugged into the MSI laptop for Motive to launch)
- **Motion capture cameras** (10 units, connected via PoE switch/modem)

> - **Calibration wand**
> - **Calibration square**
> - **Reflective markers**

Reviewer 2: ```suggestion
- **Calibration wand**
- **Calibration square**
- **Reflective markers**
```

Can we add photos for these? Also the image for the usb dongle would be nice too. It's cause these are things that are easily moved and can be lost. If we need to find them, a picture should help

</details>

<details>

<summary>Network</summary>

- **Dedicated modem/router** for the motion capture system (isolated network)

> - Ethernet cables for connecting cameras, modem, and laptop

Reviewer: ```suggestion
- Ethernet cables for connecting cameras, modem, and computer
```

</details>

---

## Connecting the Isolated Network

> _Note: This step may not need to be repeated every time. If the motion capture network is already configured, skip to the next section._

1. Connect the **motion capture modem/router** to the ethernet wall ports.

> 2. Connect the **MSI laptop** to this modem via ethernet.

Reviewer: ```suggestion
2. Connect the **mocap computer** to this modem via ethernet.
```
these are just to make it more generic, since I dont trust that MSI laptop to last forever.

3. At the **main lab modem**, disconnect the **black ethernet cable** leading from the wall port, and connect the **cream coloured cable** to the wall port.

> 4. Next, unplug the **green cable** that connects the modem to the network hub, and connect the other end of the **cream coloured cable** in it's place.

Reviewer 2: We are cooked if the cable colours ever change. I think it is better to associate the cables with the devices they connect instead of just the colour

Reviewer 3: I also was going to say that the terminology 'modem' and 'network hub' are not really correct. Instead, as @JohanneMontano said, this section should be more explicit than relying on physical descriptions of cables, etc. For example, "connect the outside network to the WAN port of epsilon-x router; disconnect the switch from the epsilon-x router's LAN port; connect the robocup-x router's LAN ports to a generic wall outlet", etc. rather than "disconnect green cable from router." (Which router? Which port? LAN or WAN?)

   <Alert type="warning">

   Doing this temporarily removes ethernet access from the main lab network. Double-check that no one else is relying on the main network via ethernet before switching.

>    </Alert>

Reviewer: This might be good to put higher in the guide 😆

5. Once complete, the motion capture system is on its own isolated network. The MSI laptop and all cameras should now communicate exclusively through this network.

---

## Configuring Motive

> 1. Open the **MSI laptop** and boot into **Windows 10**.

Reviewer: ```suggestion
1. Turn on the **mocap computer** and boot into **Windows 10**.
```

   </Alert>

You have now succesfully created a rigid body. You can alter the settings of the rigid body by highlighting all the markers encompassing the rigid body and view the tab that appears in the bottom right of the screen labelled `General Settings`. Ensure that the `Streaming ID` is set to 1, and leave the others as default.

---

## Recording a Ground Truth dataset

Before we can record our ground truth dataset, we must ensure that we are receiving messages from our Motion Capture system.

### IP Configuration

We require the robot, our computer running the binary, and the mocap system to be on the same network. The desired network is **robocup-x**.

1. Connect to the robot using the portable screen and keyboards available.
2. Run `sudo ./robocupconfiguration`
3. Change the network IP address to ensure that the three systems are on the same subnet.

> 4. Change the network to `robocup-x` and configure the network.

Reviewer: This might need to be more specific. Something like
Change SSID to robocup-x
Input robocup-x password in the password field
Press c to configure and n to apply to network changes (if I'm remembering the process correctly off the top of my head)
